bootstrap/shell/shell.go

```diff
@@ -220,6 +221,51 @@ func (s *Shell) LockFile(path string, timeout time.Duration) (LockFile, error) {
 	return &lock, err
 }
 
+func (s *Shell) flock(path string, timeout time.Duration) (LockFile, error) {
+	absolutePathToLock, err := filepath.Abs(path)
```

Comment:
That's a good idea, could be dangerous if the two libraries step over each other. I'm not sure of an alternative unless instead of requesting a specific path the users of Shell.LockFile were to specify a use for their lock from an enum instead but I don't really know if that's better in terms of abstraction. I updated the code to match what you have here and updated the test to handle it.